src/reporting/clients/ReportsClient.ts

```diff
@@ -0,0 +1,141 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { AccessToken } from "@itwin/core-bentley";
+import { RequiredError } from "../interfaces/Errors";
+import { EntityListIterator } from "../iterators/EntityListIterator";
+import { EntityListIteratorImpl } from "../iterators/EntityListIteratorImpl";
+import { Collection, getEntityCollectionPage } from "../iterators/IteratorUtil";
+import { OperationsBase } from "../OperationsBase";
+import { Report, ReportCollection, ReportSingle, ReportCreate, ReportUpdate, ReportMapping, ReportMappingCollection, ReportMappingCreate, ReportMappingSingle } from "../interfaces/Reports";
+import { IReportsClient } from "./IReportsClient";
+
+export class ReportsClient extends OperationsBase implements IReportsClient{
+  public async getReports(accessToken: AccessToken, projectId: string, top?: number): Promise<Report[]> {
+    const reports: Array<Report> = [];
+    const reportIterator = this.getReportsIterator(accessToken, projectId, top);
+    for await(const report of reportIterator) {
+      reports.push(report);
+    }
+    return reports;
+  }
+
+  public getReportsIterator(accessToken: AccessToken, projectId: string, top?: number): EntityListIterator<Report> {
+    let url = `${this.basePath}/reports?projectId=${encodeURIComponent(projectId)}&deleted=false`;
```

Comment:
The value of `deleted` query parameter should be coming from an optional function argument.

Reply:
both getReports and getReportsIterator now accept the optional parameter deleted and it is used